pkg/kudoctl/util/repo/repo.go

```diff
@@ -1,15 +1,123 @@
 package repo
 
-import "github.com/kudobuilder/kudo/pkg/apis/kudo/v1alpha1"
+import (
+	"fmt"
+	"os"
+
+	"github.com/kudobuilder/kudo/pkg/apis/kudo/v1alpha1"
+	"github.com/spf13/afero"
+	"sigs.k8s.io/yaml"
+)
+
+const (
+	// Version is the repo / packaging version
+	Version         = "v1"
+	defaultRepoName = "community"
+)
 
 // RepositoryConfiguration represents a collection of parameters for operator repository.
 type RepositoryConfiguration struct {
-	URL string `json:"url"`
+	URL  string `json:"url"`
+	Name string `json:"name"`
+}
+
+// Repositories represents the repositories.yaml file usually in the $KUDO_HOME
+type Repositories struct {
+	RepoVersion  string                     `json:"repoVersion"`
+	Context      string                     `json:"context"`
+	Repositories []*RepositoryConfiguration `json:"repositories"`
 }
 
 // Default initialized repository.
 var Default = &RepositoryConfiguration{
-	URL: "https://kudo-repository.storage.googleapis.com",
+	Name: defaultRepoName,
+	URL:  "https://kudo-repository.storage.googleapis.com",
+}
+
+// NewRepoFile creates a new repo with only defaults populated
+func NewRepoFile() *Repositories {
```

Comment:
I would actually separate (code-wise) handling of local settings and files and working with remote repositories. I think those are two different responsibilities - you have to figure out what repo to use and you do it from setting. Then you finally know and you can read from that repository. I would keep all code around local settings and setting up that in one place...